### PR TITLE
Replace auth code with auth library

### PR DESCRIPTION
**Pull request by humaton**
Replace authentication code by shared library

### CODE REVIEW

**Comment by sunilk747** [comment]
@humaton Did you test it locally. Is it working for you?

**Comment by humaton** [comment]
@sunilk747 I did run tests.

**Comment by sunilk747** [comment]
@humaton I fetch your PR and tried testing it locally, for me it's not working and also I am not getting a proper error message with status code. 
llllllll

**Comment by humaton** [comment]
The weirdest thing is that tests are passing.

**Comment by humaton** [comment]
So it seems that the problem is somewhere else. I can run the project locally and api response is:
 {
"status": "ok"
}

**Comment by sunilk747** [comment]
DId you try this endpoint `license-recommender` because this is the one which uses auth service.

**Comment by humaton** [comment]
Yes and it seems that error handling in this project is kicking in. If I check logs I can see 401 but the API returns 500

**Comment by humaton** [comment]
[TEST]

**Comment by sunilk747** [comment]
[test]

**Comment by humaton** [comment]
@sunilk747 can you please retest this?

**Comment by sunilk747** [comment]
Now I can see proper error message for invalid token.

File: run-api-local.sh [hunk 1]
```diff
@@ -34,6 +44,9 @@ while getopts ht:p: opt; do
       ;;
   esac
 done
+#setup auth library variables
+export OSIO_AUTH_URL='http://auth.openshift.io'
+export FABRIC8_ANALYTICS_JWT_AUDIENCE='fabric8-online-platform'
```
[the remark this answers is not in the document]
Comment (humaton) [reply run-api-local.sh]: Yes, we can, for example, its part of fabric8-auth-tests and other fabric8 repositories.

**Pull request merged**